pkg/controller/perconaservermongodb/secrets.go

```diff
@@ -158,3 +134,42 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsersSecret(ctx context.Context
 
 	return nil
 }
+
+func fillSecretData(cr *api.PerconaServerMongoDB, data map[string][]byte) (changes bool, err error) {
+	if data == nil {
+		data = make(map[string][]byte)
+	}
+	userMap := map[string]string{
+		envMongoDBBackupUser:         string(roleBackup),
+		envMongoDBClusterAdminUser:   string(roleClusterAdmin),
+		envMongoDBClusterMonitorUser: string(roleClusterMonitor),
+		envMongoDBUserAdminUser:      string(roleUserAdmin),
+	}
+	passKeys := []string{
+		envMongoDBClusterAdminPassword,
+		envMongoDBUserAdminPassword,
+		envMongoDBBackupPassword,
+		envMongoDBClusterMonitorPassword,
+	}
+	if cr.CompareVersion("1.13.0") >= 0 {
+		userMap[envMongoDBDatabaseAdminUser] = string(roleDatabaseAdmin)
+		passKeys = append(passKeys, envMongoDBDatabaseAdminPassword)
+	}
+	for k, defaultValue := range userMap {
+		if _, ok := data[k]; !ok {
+			data[k] = []byte(defaultValue)
+			changes = true
+		}
+	}
```

Comment:
I suggest the following for clarity:
```suggestion
	for user, role := range userMap {
		if _, ok := data[user]; !ok {
			data[user] = []byte(role)
			changes = true
		}
	}
```

Reply:
Thanks for sharing